src/main/java/core/basesyntax/Figure.java

```diff
@@ -0,0 +1,15 @@
+package core.basesyntax;
+
+public abstract class Figure implements Drawable, AreaCalculator {
+
+    protected String color;
+    private ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure() {
+        this.color = colorSupplier.getRandomColor();
+    }
```

Comment:
You can't set random color every time you create new Figure without params. Delete `colorSupplier` and remove this constructor.